griptape/events/event_bus.py

```diff
@@ -18,7 +20,11 @@ class _EventBus(SingletonMixin):
 
     @property
     def event_listeners(self) -> list[EventListener]:
-        return self._event_listeners
+        # Check if there are active listeners in the context
+        active_listeners = _active_listeners.get()
+
+        # Use active listeners if any exist, otherwise use global listeners
+        return active_listeners if active_listeners else self._event_listeners
```

Comment:
I do not like this logic

Reply:
What if I want to clear the event listeners from with a thread/coroutine? (I actually had a use case for this before)